src/Onda.jl

```diff
@@ -31,156 +30,44 @@ export Samples, encode, encode!, decode, decode!, load, store
-                                  sample_rate=signal["sample_rate"]))
-        end
-        for annotation in raw["annotations"]
-            push!(annotations, Annotation(; recording,
-                                          id=uuid_from_annotation(annotation),
-                                          span=TimeSpan(annotation["start_nanosecond"], annotation["stop_nanosecond"]),
-                                          value=annotation["value"]))
-        end
-    end
-    signals_file_path = joinpath(dataset_path, "upgraded.onda.signals.arrow")
-    verbose && log("writing out $signals_file_path...")
-    write_signals(signals_file_path, signals; kwargs...)
-    verbose && log("$signals_file_path written.")
-    annotations_file_path = joinpath(dataset_path, "upgraded.onda.annotations.arrow")
-    verbose && log("writing out $annotations_file_path...")
-    write_annotations(annotations_file_path, annotations; kwargs...)
-    verbose && log("$annotations_file_path written.")
-    return signals, annotations
+function _deprecated_read_table(io_or_path, schema=nothing)
+    table = Legolas.read_arrow(io_or_path)
+    schema isa Legolas.Schema && Legolas.validate(table, schema)
+    return table
 end
 
-_v0_4_json_value_from_annotation(ann) = JSON3.write(Dict(name => getproperty(ann, name) for name in propertynames(ann)
-                                                         if !(name in (:recording, :span))))
-
-"""
-    downgrade_onda_dataset_to_v0_4!(dataset_path, signals, annotations;
-                                    verbose=true,
-                                    value_from_annotation=Onda._v0_4_json_value_from_annotation,
-                                    signal_file_extension_and_options_from_format=(fmt -> (fmt, nothing)))
-
-Write an Onda-Format-v0.4-compliant `recordings.msgpack.zst` file to `dataset_path` given Onda-Format-v0.5-compliant
-`signals` and `annotations` tables.
-
-- This function internally uses `Onda.gather`, and thus expects `signals`/`annotations` to support `view` for
-  row extraction. One way to ensure this is the case is to convert `signals`/`annotations` to `DataFrame`s before
-  passing them to this function.
-
-- If `verbose` is `true`, this function will print out timestamped progress logs.
-
-- `value_from_annotation` is a function that takes in an `Onda.Annotation` and returns the string that
-  should be written out as that annotation's value. By default, this value will be a JSON object string
-  whose fields are all fields in the given annotation except for `recording` and `span`.
-
-- `signal_file_extension_and_options_from_format` is a function that takes in `signal.file_format` and
-  returns the `file_extension` and `file_options` fields that should be written out for the signal.
+function read_signals(io_or_path; validate_schema::Bool=true)
+    @warn "`Onda.read_signals(io_or_path)` is deprecated, use `Legolas.read(io_or_path)` instead"
```

Comment:
Good tips.

I didn't use `@deprecate` because the method I'm "redirecting to" for the deprecation path (`_deprecated_read_table`) is different than the method that I'm suggesting downstream callers use instead (`Legolas.read`).

AFAICT `Base.depwarn` isn't officially a documented function, so I was meh on using it here, though it's probably fine in practice.

I think I'll go with `maxlog=1`